include/COSMO_SAC/profile_db.hpp

```diff
@@ -346,6 +347,90 @@ class DelawareProfileDatabase : public ProfileDatabase {
     }
 };
 
+/// The DirectImport feature for importing profiles based on "filename" + "path" 
+class DirectImport : public ProfileDatabase {
+private:
+    std::string m_dbpath;
+
+public:
+    DirectImport() {
+    }
+
+    void add_profile(const std::string& identifier, const std::string& dbpath = ".") {
+        m_dbpath = dbpath;
+        // Now we load the sigma profile(s) from the file
+        auto lines = str_split(get_file_contents(m_dbpath + "/" + identifier + ".sigma"));
+
+        std::vector<double> _sigma, _psigmaA;
+        FluidProfiles fluid;
+        std::string meta;
+        for (auto &&line : lines) {
+            if (line.substr(0,8) == "# Name: ") { 
+                std::string check_name = line.substr(8); 
+                continue;
+            }
+            if (line.substr(0,8) == "# CASn: ") { 
+                std::string check_CAS = line.substr(8); 
+                continue; 
+            }
+            if (line.substr(0,8) == "# meta: ") { 
+                meta = line.substr(8); 
+                continue; 
+            }
+            if (line[0] == '#'){ continue; }
+            auto v = str_split(strrstrip(line), " ");
+            if (v.empty() || (v.size() == 1 && v[0].empty())) { continue; }
+            _sigma.push_back(mystrtod(v[0]));
+            _psigmaA.push_back(mystrtod(v[1]));
+        }
```

Comment:
@ianhbell Thanks for pointing this out! I have removed the unnecessary second argument from the `substr` method. The code is now simplified.